modules/images/image-loading-optimization/detection/detect.js

```diff
@@ -84,22 +132,76 @@ function getBreadcrumbs( leafElement ) {
 	return breadcrumbs;
 }
 
+/**
+ * Checks whether the page metric(s) for the provided viewport width is needed.
+ *
+ * @param {number}                   viewportWidth               - Current viewport width.
+ * @param {Array<number, boolean>[]} neededMinimumViewportWidths - Needed minimum viewport widths, in ascending order.
+ * @return {boolean} Whether page metrics are needed.
+ */
+function isViewportNeeded( viewportWidth, neededMinimumViewportWidths ) {
+	let lastWasNeeded = false;
+	for ( const [
+		minimumViewportWidth,
+		isNeeded,
+	] of neededMinimumViewportWidths ) {
+		if ( viewportWidth >= minimumViewportWidth ) {
+			lastWasNeeded = isNeeded;
+		} else {
+			break;
+		}
+	}
+	return lastWasNeeded;
+}
+
+/**
+ * Gets the current time in milliseconds.
+ *
+ * @return {number} Current time in milliseconds.
+ */
+function getCurrentTime() {
+	return new Date().valueOf();
+}
```

Comment:
To prevent issues with timezones and other kinds of clock adjustments, WDYT about using `performance.now()` instead?

Reply:
Also, `performance.now()` returns not the current time but the current time in milliseconds since the start of the page load.

Reply:
> Also, `performance.now()` returns not the current time but the current time in milliseconds since the start of the page load.

Which is fine because from what I can see you only need relative time anyway for the lock, no?

Reply:
The lock is captured at a specific `time()` not from the start of the page load time. When the REST API request to update page metrics is successful, it sets a lock for the IP at the current time. So we need to compare against that time.

I've refined time handling a bit in 6cd8198532b7076770d93b2003364949d0d3030f.